pybamm_tea/tea.py

```diff
@@ -333,209 +333,102 @@ def print_stack_breakdown(self):
     def print_stack_energy_densities(self):
         """A dataframe with capacities, energy densities, stoichiometry- and potential windows, n/p ratios, (single-)stack thickness and stack density."""
         stack_ed = self.stack_energy_densities
-        data = [
-            {
-                "Parameter": "Volumetric energy density",
-                "Unit": "Wh.L-1",
-                "Value": stack_ed.get("Volumetric stack energy density [Wh.L-1]"),
-            },
-            {
-                "Parameter": "Gravimetric energy density",
-                "Unit": "Wh.kg-1",
-                "Value": stack_ed.get("Gravimetric stack energy density [Wh.kg-1]"),
-            },
-            {
-                "Parameter": "Stack average OCP",
-                "Unit": "V",
-                "Value": stack_ed.get("Stack average OCP [V]"),
-            },
-            {
-                "Parameter": "Capacity",
-                "Unit": "mA.h.cm-2",
-                "Value": stack_ed.get("Capacity [mA.h.cm-2]"),
-            },
-            {
-                "Parameter": "(Single-) stack thickness",
-                "Unit": "um",
-                "Value": 10**6 * stack_ed.get("Stack thickness [m]"),
-            },
-            {
-                "Parameter": "Stack density",
-                "Unit": "kg.L-1",
-                "Value": 10**-3 * stack_ed.get("Stack density [kg.m-3]"),
-            },
-        ]
-        # Create the DataFrame from the list of dictionaries
+
+        data = {
+            "Parameter": [
+                "Volumetric energy density",
+                "Gravimetric energy density",
+                "Stack average OCP",
+                "Capacity",
+                "(Single-) stack thickness",
+                "Stack density",
+            ],
+            "Unit": [
+                "Wh.L-1",
+                "Wh.kg-1",
+                "V",
+                "mA.h.cm-2",
+                "um",
+                "kg.L-1",
+            ],
+            "Value": [
+                stack_ed.get("Volumetric stack energy density [W.h.L-1]"),
+                stack_ed.get("Gravimetric stack energy density [W.h.kg-1]"),
+                stack_ed.get("Stack average OCP [V]"),
+                stack_ed.get("Capacity [mA.h.cm-2]"),
+                10**6 * stack_ed.get("Stack thickness [m]"),
```

Comment:
Just add "Stack thickness [um]" and "Stack density [kg.L-1]" to `stack_ed` too when you construct it, then you can just do the loop here too